envisage/provider_extension_registry.py

```diff
@@ -42,7 +42,7 @@ class ProviderExtensionRegistry(ExtensionRegistry):
     def set_extensions(self, extension_point_id, extensions):
         """ Set the extensions to an extension point. """
 
-        raise SystemError("extension points cannot be set")
+        raise RuntimeError("extension points cannot be set")
```

Comment:
I'd suggest `TypeError` here - that seems like the most appropriate thing for an operation which isn't supported on a particular type.
```suggestion
        raise TypeError("extension points cannot be set")
```